src/app-root.ts

```diff
@@ -733,25 +743,50 @@ export class AppRoot extends LitElement {
   }
 
   /**
-   * Handler for when the dev panel's "Show facet top slot" checkbox is changed.
+   * Handler for when the dev panel's "Show sort bar top left slot" checkbox is changed.
    */
-  private sortBarSlotCheckboxChanged(e: Event) {
+  private sortBarLeftSlotCheckboxChanged(e: Event) {
     const target = e.target as HTMLInputElement;
 
-    const div = document.createElement('div');
-    div.style.setProperty('border', '1px solid #000');
-    div.textContent = 'Btn';
-    div.style.setProperty('height', '3rem');
-    div.style.setProperty('width', '3rem');
-    div.style.backgroundColor = '#00000';
-    div.setAttribute('slot', 'sort-options-left');
+    if (target.checked) {
+      const div = document.createElement('div');
+      div.style.setProperty('border', '1px solid #000');
+      div.textContent = 'Btn';
+      div.style.setProperty('height', '3rem');
+      div.style.setProperty('width', '3rem');
+      div.style.backgroundColor = '#000';
+      div.setAttribute('slot', 'sort-options-left');
+
+      this.collectionBrowser.appendChild(div);
+    } else {
+      const slottedEl = this.collectionBrowser.querySelector(
+        '[slot="sort-options-left"]'
+      );
+      if (slottedEl) this.collectionBrowser.removeChild(slottedEl);
+    }
+  }
+
+  /**
+   * Handler for when the dev panel's "Show sort bar top right slot" checkbox is changed.
+   */
+  private sortBarRightSlotCheckboxChanged(e: Event) {
+    const target = e.target as HTMLInputElement;
 
     if (target.checked) {
+      const div = document.createElement('div');
+      div.style.setProperty('border', '1px solid #000');
+      div.textContent = 'Search bar';
+      div.style.setProperty('height', '3rem');
+      div.style.setProperty('width', '15rem');
+      div.style.backgroundColor = '#000';
+      div.setAttribute('slot', 'sort-options-right');
+
       this.collectionBrowser.appendChild(div);
     } else {
-      this.collectionBrowser.removeChild(
-        this.collectionBrowser.lastElementChild as Element
+      const slottedEl = this.collectionBrowser.querySelector(
+        '[slot="sort-options-right"]'
```

Comment:
Cleaned up some dubious logic in these demo app methods. Since multiple elements might be slotted in at once, we can't rely on the last element being the one that should be removed.